### PR TITLE
chore: Improve name and signature of focus hook

### PR DESCRIPTION
**Pull request by VincentSmedinga**
This renames the hook to ‘use focus with arrows’ to ‘use keyboard focus’ because we support more keys than only the arrows to manage focus.

It changes the export type from default to named.

And I’ve moved the `ref` parameter out of the options object. That object was added because of a request of mine in a pull request, but I had intended only the Boolean configuration options to be in it, not the ref object.

We don’t export this function, so I don’t consider it a breaking change, correct?

### CODE REVIEW

File: packages/react/src/common/useFocusByKeyboard.tsx [hunk 1]
```diff
@@ -24,21 +24,18 @@ const FOCUSABLE_ELEMENTS = [
...
   horizontally?: boolean
 }
 
-const useFocusWithArrows = ({
-  ref,
-  rotating = false,
-  directChildrenOnly = false,
-  horizontally = false,
-}: FocusWithArrowsOptions) => {
+export const useFocusByKeyboard = (ref: RefObject<HTMLDivElement>, options: FocusByKeyboardOptions) => {
```
Comment (VincentSmedinga) [inline packages/react/src/common/useFocusByKeyboard.tsx:33]: Wait… is it correct that we use `HTMLDivElement` here? Why div?

Comment (alimpens) [inline packages/react/src/common/useFocusByKeyboard.tsx:33]: I like `useKeyboardFocus`!

Comment (alimpens) [reply packages/react/src/common/useFocusByKeyboard.tsx:33]: No, I don't think so, should be `HTMLElement` I think

File: packages/react/src/common/useFocusByKeyboard.tsx [hunk 2]
```diff
@@ -24,21 +24,18 @@ const FOCUSABLE_ELEMENTS = [
   'select:not([disabled])',
 ]
 
-type FocusWithArrowsOptions = {
-  ref: RefObject<HTMLDivElement>
```
Comment (alimpens) [inline packages/react/src/common/useFocusByKeyboard.tsx:28]: Why move ref from the params object? Adding it to the object documents it nicely, imo

File: packages/react/src/common/useFocusByKeyboard.tsx [hunk 3]
```diff
@@ -24,21 +24,18 @@ const FOCUSABLE_ELEMENTS = [
   'select:not([disabled])',
```
Comment (alimpens) [inline packages/react/src/common/useFocusByKeyboard.tsx:24]: Maybe not for now, but: do we want to document this hook? We expose it to our users already

File: packages/react/src/common/useFocusByKeyboard.tsx (hunk 1 above)
Comment (alimpens) [reply packages/react/src/common/useFocusByKeyboard.tsx]: Hm, does changing it to `HTMLElement` cause TypeScript errors? I think maybe the 'correct' way to do this is to [use a generic and pass the type to the function when calling it](https://stackoverflow.com/questions/57201223/typescript-how-to-declare-a-type-that-includes-all-types-extending-a-common-typ#answer-57201395), but that might be a bit much for now.

File: packages/react/src/common/useFocusByKeyboard.tsx (hunk 3 above)
Comment (alimpens) [reply packages/react/src/common/useFocusByKeyboard.tsx]: Ah, sorry, I was wrong. It is a function in `index.esm.js` and `index.cjs.js`, but it's not exported, so you probably can't reach it?

Comment (VincentSmedinga) [reply packages/react/src/common/useFocusByKeyboard.tsx]: Yes, nothing seems to indicate it’s public.

File: packages/react/src/common/useFocusByKeyboard.tsx (hunk 1 above)
Comment (VincentSmedinga) [reply packages/react/src/common/useFocusByKeyboard.tsx]: I think I’ve applied that correctly; can you verify?

Comment (alimpens) [reply packages/react/src/common/useFocusByKeyboard.tsx]: Hm, I don't this this is any different from what we had? `ref: RefObject<HTMLElement>` seems to work for all types of html elements, and `<T extends HTMLElement>(ref: RefObject<T>)` seems to do the same thing.

The first option is good enough imo, and a bit simpler than the second one.

Comment (VincentSmedinga) [reply packages/react/src/common/useFocusByKeyboard.tsx]: I did get the TS error I quoted, but not anymore now. Maybe some other change. Reverted the generic.

**Review by alimpens — APPROVED** [review]

**Pull request merged**